packages/arb-token-bridge-ui/src/util/WithdrawalUtils.ts

```diff
@@ -9,52 +9,60 @@ import { Address } from './AddressUtils'
 export async function withdrawInitTxEstimateGas({
   amount,
   address,
-  l2Provider,
+  childChainProvider,
   erc20L1Address
 }: {
   amount: BigNumber
   address: Address
-  l2Provider: Provider
+  childChainProvider: Provider
   erc20L1Address?: string
 }): Promise<GasEstimates> {
   const isToken = typeof erc20L1Address === 'string'
-  const bridger = await (isToken ? Erc20Bridger : EthBridger).fromProvider(
-    l2Provider
-  )
-
-  const withdrawalRequest = isToken
-    ? await (bridger as Erc20Bridger).getWithdrawalRequest({
-        amount,
-        destinationAddress: address,
-        from: address,
-        erc20l1Address: erc20L1Address
-      })
-    : await (bridger as EthBridger).getWithdrawalRequest({
-        amount,
-        destinationAddress: address,
-        from: address
-      })
+  const estimatedChildChainSubmissionCost = constants.Zero
 
   // For the withdrawal init tx, there's no gas fee paid on L1 separately
   // unless we break down part of the L2 gas fee into L2 tx fee + L1 batch posting fee for that tx
   // but as that does not fit into the context of the transfer. and
   // would lead to user confusion, we instead present all the gas fee as L2 gas fee
-  const estimatedL1Gas = BigNumber.from(0)
+  const estimatedParentChainGas = BigNumber.from(0)
 
   try {
+    const bridger = await (isToken ? Erc20Bridger : EthBridger).fromProvider(
+      childChainProvider
+    )
+
+    const withdrawalRequest = isToken
+      ? await (bridger as Erc20Bridger).getWithdrawalRequest({
+          amount,
+          destinationAddress: address,
+          from: address,
+          erc20l1Address: erc20L1Address
+        })
+      : await (bridger as EthBridger).getWithdrawalRequest({
+          amount,
+          destinationAddress: address,
+          from: address
+        })
```

Comment:
If we wrap it with `if (isToken) { const bridger = ...}`, we don't need to cast anymore